microprofile/telemetry/src/main/java/io/helidon/microprofile/telemetry/HelidonTelemetryContainerFilter.java

```diff
@@ -92,13 +93,18 @@ public Iterable<String> keys(ContainerRequestContext containerRequestContext) {
     HelidonTelemetryContainerFilter(Tracer tracer, OpenTelemetry openTelemetry, org.eclipse.microprofile.config.Config mpConfig) {
         this.tracer = tracer;
         this.openTelemetry = openTelemetry;
+        isAgentPresent = Boolean.getBoolean(TelemetryCdiExtension.OTEL_AGENT_PRESENT);
```

Comment:
Why didn't you use `io.helidon.tracing.providers.opentelemetry.HelidonOpenTelemetry.AgentDetector`?